cmd/web-handlers.go

```diff
@@ -667,10 +667,13 @@ next:
 	for _, objectName := range args.Objects {
 		// If not a directory, remove the object.
 		if !HasSuffix(objectName, SlashSeparator) && objectName != "" {
-			// Check for permissions only in the case of
-			// non-anonymous login. For anonymous login, policy has already
-			// been checked.
-			govBypassPerms := ErrAccessDenied
+			// Get the retention/lock info to check if it is enabled or not
+			retentionInfo, _ := globalBucketObjectLockConfig.Get(args.BucketName)
```

Comment:
you need to only know 


```
info, ok := globalBucketObjectLockConfig.Get(args.BucketName)
if ok {
}
```
is true.

Reply:
@harshavardhana 

There is a difference between `if info.Mode != "" {` and `if ok {`.

After a user `--clear`s an existing lock configuration, the configuration is emptied, but it is still there, so `ok` returns `true` while `Mode=""`. So, there won't be any lock/retention enforcement on the bucket and if we choose to depend on `ok`, then we'll go into checking the retention/lock actions. When the lock is disabled, That's why I chose to not depend on the `ok`, bu the Mode info instead.